sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusScalerProvider.cs

```diff
@@ -0,0 +1,104 @@
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Extensions.ServiceBus.Listeners
+{
+    internal class ServiceBusScalerProvider : IScaleMonitorProvider, ITargetScalerProvider
+    {
+        private readonly ServiceBusScaleMonitor _scaleMonitor;
+        private readonly ServiceBusTargetScaler _targetScaler;
+
+        public ServiceBusScalerProvider(IServiceProvider serviceProvider, TriggerMetadata triggerMetadata)
+        {
+            AzureComponentFactory azureComponentFactory;
+            if ((triggerMetadata.Properties != null) && (triggerMetadata.Properties.TryGetValue(nameof(AzureComponentFactory), out object value)))
+            {
+                azureComponentFactory = value as AzureComponentFactory;
+            }
+            else
+            {
+                azureComponentFactory = serviceProvider.GetService<AzureComponentFactory>();
+            }
+
+            IConfiguration configuration = serviceProvider.GetService<IConfiguration>();
+            MessagingProvider messagingProvider = serviceProvider.GetService<MessagingProvider>();
+            AzureEventSourceLogForwarder logForwarder = serviceProvider.GetService<AzureEventSourceLogForwarder>();
+            IOptions<ServiceBusOptions> options = serviceProvider.GetService<IOptions<ServiceBusOptions>>();
+            ILoggerFactory loggerFactory = serviceProvider.GetService<ILoggerFactory>();
+            ServiceBusMetadata serviceBusMetadata = JsonConvert.DeserializeObject<ServiceBusMetadata>(triggerMetadata.Metadata.ToString());
+            ServiceBusClientFactory factory = new ServiceBusClientFactory(configuration, azureComponentFactory, messagingProvider, logForwarder, options);
+            ServiceBusAdministrationClient serviceBusAdminClient = factory.CreateAdministrationClient(serviceBusMetadata.Connection);
+            ServiceBusClient serviceBusClient = factory.CreateClientFromSetting(serviceBusMetadata.Connection);
+            ServiceBusReceiver serviceBusReceiver = !string.IsNullOrEmpty(serviceBusMetadata.QueueName) ? serviceBusClient.CreateReceiver(serviceBusMetadata.QueueName)
+            : serviceBusClient.CreateReceiver(serviceBusMetadata.TopicName, serviceBusMetadata.SubscriptionName);
+
+            _scaleMonitor = new ServiceBusScaleMonitor(
+                triggerMetadata.FunctionName,
+                serviceBusReceiver.EntityPath,
+                !string.IsNullOrEmpty(serviceBusMetadata.QueueName) ? ServiceBusEntityType.Queue : ServiceBusEntityType.Topic,
+                new Lazy<ServiceBusReceiver>(() => serviceBusReceiver),
+                new Lazy<ServiceBusAdministrationClient>(() => serviceBusAdminClient),
+                loggerFactory);
+
+            _targetScaler = new ServiceBusTargetScaler(
```

Comment:
same here:  should this be removed - https://github.com/Azure/azure-sdk-for-net/blob/aa1ef2a332d38a95e7c7fd39a7c8659754b63f19/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusListener.cs#L115

Reply:
No, there a re 2 different paths to create `ServiceBusTargetScaler`- one is regular trigger creation and another one is creation from SC.